### ISSUE TITLE
Compare calldata requirements for designs 2 & 3

### ISSUE THREAD

**Issue by andrewgordstewart** [statechannels/dispute-game#9]
Ignore the storage cost of each round -- assume we can store an array of arbitrary size for free. (This will be tackled later).

Assume a maximum of 132KB for calldata, and assume that any "cryptographically important" word is 32 bytes.

Answer the following question, by deriving how many "words" need to be supplied to perform a single k-section.
> How much k-section can be done in one turn in [option 2](https://www.notion.so/statechannels/Various-bisection-games-8d5ed428f7114b52a1c6bbeb08fe96d5#38295eb9de974acba1853d35d9c76697) vs the [symmetric version of option 3](https://www.notion.so/statechannels/Various-bisection-games-8d5ed428f7114b52a1c6bbeb08fe96d5#845e716c399547f2971a98247e0318f6)?

This forms a part of the justification of option 3 over option 2, relevant to _**[maximizing the feasible branching factor](https://www.notion.so/statechannels/Optimism-SoW-474c48fb77334742a86f4b508912e85e#747224267be24fe09530645afa6113a9)**_ from Milestone 3.

**Comment by lalexgap** [comment]
I've created this [document](https://www.notion.so/statechannels/Dispute-Games-Call-Data-Size-416a65c226054aba8c89d04c7fd3f380) explaining the call data size:

TLDR: Both approaches will have a call data size of `log_2(k) + k + c`

**Comment by andrewgordstewart** [comment]
> I've created this [document](https://www.notion.so/statechannels/Dispute-Games-Call-Data-Size-416a65c226054aba8c89d04c7fd3f380) explaining the call data size:
> 
> TLDR: Both approaches will have a call data size of `log_2(k) + k + c`

Note: an eager over-optimizer could reduce the calldata size for Option 3 (we call it array dissection?) to `k + c`, by using eg. an RSA accumulator to "store" an array `a` of length `k`, rather than a Merkle tree.

(I haven't thought carefully about whether you could get rid of the `log_2(k)` cost in Option 2, but you probably can.)
